apps/web/src/services/iam/components/RoleUpdateFormAccess.vue

```diff
@@ -1,53 +1,70 @@
 <script setup lang="ts">
-import { computed, reactive, watch } from 'vue';
+import {
+    computed, reactive, ref, watch,
+} from 'vue';
+import { useRouter } from 'vue-router/composables';
 
-import { filter } from 'lodash';
+import { filter, find, isEqual } from 'lodash';
 
 import {
     PI, PToggleButton, PDataTable, PSelectDropdown, PCheckboxGroup, PCheckbox, PHeading,
+    useProxyValue,
 } from '@cloudforet/mirinae';
 
 import { ROLE_TYPE } from '@/api-clients/identity/role/constant';
-import type { RoleType } from '@/api-clients/identity/role/type';
 import { i18n } from '@/translations';
 
 import { PAGE_ACCESS } from '@/lib/access-control/config';
+import { getPageAccessMapFromRawData } from '@/lib/access-control/page-access-helper';
 import { MENU_ID } from '@/lib/menu/config';
 
 import { gray, green } from '@/styles/colors';
 
-import type { PageAccessMenuItem, TableItem } from '@/services/iam/types/role-type';
+import { useRoleBasedMenus } from '@/services/iam/composables/role-based-menus';
+import { useRoleGetQuery } from '@/services/iam/composables/use-role-get-query';
+import { getPageAccessList, getPageAccessMenuListByRoleType } from '@/services/iam/helpers/role-page-access-menu-list';
+import type { PageAccessMenuItem, RoleFormData, TableItem } from '@/services/iam/types/role-type';
+
 
 interface Props {
     menuItems?: PageAccessMenuItem[]
-    roleType?: RoleType
+    proxyPageAccessValid?: boolean;
 }
 
-const props = withDefaults(defineProps<Props>(), {
-    menuItems: undefined,
-    roleType: ROLE_TYPE.DOMAIN_ADMIN,
-});
+const props = defineProps<Props>();
 
-const emit = defineEmits<{(e: 'update', value: PageAccessMenuItem): void,
+const emit = defineEmits<{(e: 'update', value: string[]): void,
+    (e: 'update-form', formData: RoleFormData): void,
 }>();
 
+const router = useRouter();
+const roleId: string = router.currentRoute.params.id;
+
+const { roleData } = useRoleGetQuery(roleId);
+
+const { roleBasedMenus } = useRoleBasedMenus();
+
+const menuItems = ref([] as PageAccessMenuItem[]);
 const state = reactive({
-    menuItems: [] as PageAccessMenuItem[],
-    accessibleMenuList: computed<PageAccessMenuItem[]>(() => state.menuItems.flatMap((i) => i.subMenuList)),
+    pageAccessPermissions: computed(() => getPageAccessList(menuItems.value)),
+    proxyPageAccessValid: useProxyValue('proxyPageAccessValid', props, emit),
```

Comment:
suggestion: Since `useProxyValue` internally performs emits in this manner, it's better to add it to the `emit` type.

`useProxyValue`는 내부적으로 emit을 이런식으로 실행하기 때문에, emit 타입에 추가해주는 게 좋습니다.

```typescript
emit(`update:${name}`, value);
emit(`update:${kebabCaseName}`, value);
```


```
const emit = defineEmits<{(e: 'update', value: string[]): void,
    (e: 'update-form', formData: RoleFormData): void,
    (event: 'update:page-access-valid', val: boolean): void;
}>();
```